pkg/manager/member/orphan_pods_cleaner.go

```diff
@@ -89,6 +110,23 @@ func (opc *orphanPodsCleaner) Clean(tc *v1alpha1.TidbCluster) (map[string]string
 			return skipReason, err
 		}
 
+		// check api server
+		_, err = opc.kubeCli.CoreV1().PersistentVolumeClaims(ns).Get(pvcName, metav1.GetOptions{})
+		if err == nil {
+			skipReason[podName] = skipReasonOrphanPodsCleanerPVCIsFound
+			continue
+		}
+		if !errors.IsNotFound(err) {
+			return skipReason, err
+		}
+
+		// if the PVC is not found in apiserver (also informer cache) and the
+		// phrase of the Pod is Pending, delete it and let the stateful
+		// controller to create the pod and its PVC(s) again
+		if pod.Status.Phase != v1.PodPending {
```

Comment:
The pod status may also be changed during this time from pending to running.

Reply:
done in https://github.com/pingcap/tidb-operator/pull/878/commits/7a3caf0fd6fe28106ffc5c67afad53bf438f51a1